index.js

```diff
@@ -39,20 +42,65 @@ if (!fs.existsSync('./dist')) {
 }
 app.use(express.static('./dist'));
 
-app.post('/register', function(req, res) {
-  crypto.randomBytes(32, function(ex, buf) {
-    var token = buf.toString('hex');
+// load current data for
+app.get('/', function(req, res) {
+  res.render('index', { title: 'Hey', message: 'Hello there!'});
+});
 
-    // save token in database
-    client.hmset(token, {
+// adds a new machine to a token set
+// creates a new token set if needed
+app.post('/register', function(req, res) {
+  // add/update machine in database
+  var machineId = req.body.machineId;
+  new Promise(function(resolve, reject) {
+    if (!req.body.token) {
+      return resolve();
+    }
+    client.exists(req.body.token, function(err, result) {
+      if (!result || err) {
+        return reject();
+      }
+      resolve();
+    });
+  }).then(function() {
+    client.hmset(machineId, {
```

Comment:
In theory we should create machineId and token in a single transaction (using `multi`). Let's do it in a follow-up.